src/software/new_geom/polygon.cpp

```diff
@@ -2,9 +2,33 @@
 
 #include <unordered_set>
 
-Polygon::Polygon(const std::vector<Point>& points) : points_(points) {}
+Polygon::Polygon(const std::vector<Point>& points)
+    : points_(points), segments_(initSegments(points_))
```

Comment:
IIRC we used to do something like this (hold points and segments at the same time in Polygon), but decided against it because the information is redundant and is more to keep track of / update. I'd really prefer not to do this unless there is a massive performance benefit.

Reply:
pretty massive since we can't return a const reference due to the segments being constructed in the function

Reply:
1.53s vs 460ms. getSegments is called on _every_ intersect with a Polygon

Reply:
Hmmm that is a big difference. @garethellis0 @EvanMorcom @akhilveeraghanta @jsn0716 thoughts?

Reply:
@jonathanlew  has a comment that explains why we do that right after, so its fine the way it is imo, thats a huge difference for a little bit of redundant data

maybe just expand on the comment to be more specific on what it improves?

Reply:
That's pretty big. I'm surprised that calling getSegment() adds that much.

IMO, it's fine with the change.

Reply:
Yeah I'm cool with this. Optimization is neatly abstracted away by this class and there's a comment explaining why we do it. Might be worth saying _why_ this improves performance in the comment.